conans/client/graph/graph.py

```diff
@@ -23,13 +25,13 @@
 
 
 class Node(object):
-    def __init__(self, conan_ref, conanfile):
+    def __init__(self, conan_ref, conanfile, recipe=None):
         self.conan_ref = conan_ref
         self.conanfile = conanfile
         self.dependencies = []  # Ordered Edges
         self.dependants = set()  # Edges
         self.binary = None
-        self.recipe = None
+        self.recipe = recipe
```

Comment:
Probably not the best name, but no other suggestion.

Reply:
Yes, same. Don't like much the name, but as the field is called that...